apps/emqx/src/emqx_map_lib.erl

```diff
@@ -152,7 +152,7 @@ diff_maps(NewMap, OldMap) ->
 binary_string_kv(K, V, JsonableFun) ->
     case JsonableFun(K, V) of
         drop -> drop;
-        {K1, V1} -> {binary_string(K1), binary_string(V1)}
+        {K1, V1} -> {binary_string(K1), V1}
```

Comment:
Not really.  for example integer:
```erlang
3> jiffy:encode(#{<<"test">> => "1"}).
<<"{\"test\":[49]}">>
4> jiffy:encode(#{<<"test">> => 1}).
<<"{\"test\":1}">>
5> jiffy:encode(#{<<"test">> => <<"1">>}).
<<"{\"test\":\"1\"}">>
```
if we change all values to binary, it will become a string in JSON, lost the origin type(int,array,boolean etc).
such as:
`cluster.mcast.ports` is array.
```yaml
cluster {
  mcast {    
    ## @doc cluster.mcast.ports
    ## ValueType: Array<Port>
    ## Default: [4369, 4370]
    ports = [4369, 4370]
```
we should return `"ports":[4369,4370]` not `"ports":"ᄑᄒ"`

Reply:
OK. This function and the `emqx_map_lib:jsonable_map/1` is supposed to be removed later, as @zmstone said we always make sure the raw_config a JSON-able map.